data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/PercentSamplerAggregateActionConfig.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-Limense-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.aggregate.actions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+
+public class PercentSamplerAggregateActionConfig {
+    @JsonProperty("percent")
+    @NotNull
+    double percent;
+    
+    public double getPercent() {
+        if (percent <= 0.0 || percent >= 100.0) {
```

Comment:
Please replace this with a JSR validation. I'm not sure you can use `@Min`/`@Max` here. But, you can use `AssertTrue` similar to the following:

https://github.com/opensearch-project/data-prepper/blob/1b03b511a8af212dfd45f6813dee05a4a47e64c4/data-prepper-plugins/csv-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/csv/CsvProcessorConfig.java#L96-L99